backend/v1/components/aquariums/utils.py

```diff
@@ -1,44 +1,94 @@
-    if collection:
-        result = collection.insert_one(aquarium_data.dict(by_alias=True))
-        return result.inserted_id
-    return None
+def convert_mongo_id(document):
+    """
+    Converts MongoDB document _id field from ObjectId to string.
+    """
+    if "_id" in document and isinstance(document["_id"], ObjectId):
+        document["_id"] = str(document["_id"])
+    return document
+
+
+def create_aquarium(aquarium: Aquarium):
+    try:
+        connector.get_aquariums_collection().insert_one(aquarium.model_dump())
+    except Exception as e:
+        print(e)
+        print(f'Failed to create fish species: {aquarium.model_dump()}')
+        return {'code': 500, 'message': 'Failed to create fish species'}
+    return JSONResponse(content={'code': 200, 'message': 'Fish species created successfully'})
 
 
 def get_all_aquariums():
-    collection = connector.get_aquariums_collection()
-    if collection:
-        return list(collection.find({}))
-    return []
+    aquariums = connector.get_aquariums_collection().find()
+    aquariums = []
+    for s in aquariums:
+        print(s)
+        s = convert_mongo_id(s)
+        aquariums.append(s)
+    return JSONResponse(
+        content={'code': 200, 'message': 'Aquariums retrieved successfully', 'species': aquariums})
+
+
+def update_aquarium(aquarium_data: Aquarium, aquarium_id: str):
+    if not connector.get_aquariums_collection().find_one:
+        return {'code': 404, 'message': f'Aquarium {aquarium_id} not found'}
 
+    id = ObjectId(aquarium_id)
+    connector.get_aquariums_collection().find_one_and_update({'_id': id},
+                                                           {'$set': aquarium_data.model_dump()})
+    return {'code': 200,
+            'message': f'{aquarium_id} updated successfully'}
 
-def get_aquarium_by_id(aquarium_id: ObjectId):
-    collection = connector.get_aquariums_collection()
-    if collection:
-        return collection.find_one({"_id": aquarium_id})
-    return None
 
+def delete_aquarium(aquarium_id: str):
+    id = ObjectId(aquarium_id)
+    if not connector.get_aquariums_collection().find_one({'_id': id}):
+        return {'code': 404, 'message': f'Aquarium {aquarium_id} not found'}
 
-def update_aquarium(aquarium_data: Aquarium):
-    collection = connector.get_aquariums_collection()
-    if collection:
-        result = collection.update_one({"_id": aquarium_data.id}, {"$set": aquarium_data.dict(by_alias=True)})
-        return result.modified_count > 0
-    return False
+    # Delete the fish species
+    connector.get_aquariums_collection().delete_one({'_id': id})
+    return {'code': 200, 'message': f'{aquarium_id} deleted successfully'}
 
 
-def delete_aquarium(aquarium_id: ObjectId):
-    collection = connector.get_aquariums_collection()
-    if collection:
-        result = collection.delete_one({"_id": aquarium_id})
-        return result.deleted_count > 0
-    return False
+# {
```

Comment:
Let me know if this comment is necessary. If it is, then I'd suggest turning it into block comment

Reply:
comment is for easier testing to have a model ready